pkg/ebpf/c/tracee.bpf.c

```diff
@@ -3913,13 +3911,36 @@ static __always_inline bool is_socket_supported(struct socket *sock)
 
     return 1; // supported
 }
-// clang-format on
 
 //
 // network related maps
 //
 
-// clang-format off
+//
+// Type definitions and prototypes for protocol parsing
+//
+
+// NOTE: proto header structs need full type in vmlinux.h (for correct skb copy)
+
+typedef union iphdrs_t {
+    struct iphdr iphdr;
+    struct ipv6hdr ipv6hdr;
+} iphdrs;
+
+typedef union protohdrs_t {
+    struct tcphdr tcphdr;
+    struct udphdr udphdr;
+    struct icmphdr icmphdr;
+    struct icmp6hdr icmp6hdr;
+    union {
+        u8 tcp_extra[40]; // data offset might set it up to 60 bytes
+    };
+} protohdrs;
+
+typedef struct nethdrs_t {
+    iphdrs iphdrs;
+    protohdrs protohdrs;
```

Comment:
Positional change.

Reply:
All struct definitions are now under types.h, so better move it there or to network.h